src/platform/packages/private/kbn-grid-layout/grid/use_grid_layout_events/panel/state_manager_actions.ts

```diff
@@ -8,134 +8,152 @@
  */
 
 import { cloneDeep } from 'lodash';
-import deepEqual from 'fast-deep-equal';
 import { MutableRefObject } from 'react';
-import { GridLayoutStateManager, GridPanelData } from '../../types';
-import { getDragPreviewRect, getSensorOffsets, getResizePreviewRect } from './utils';
-import { resolveGridRow } from '../../utils/resolve_grid_row';
-import { isGridDataEqual } from '../../utils/equality_checks';
-import { PointerPosition, UserInteractionEvent } from '../types';
+
+import {
+  ActivePanelEvent,
+  GridLayoutStateManager,
+  GridPanelData,
+  GridSectionData,
+  OrderedLayout,
+} from '../../types';
+import { GridLayoutContextType } from '../../use_grid_layout_context';
+import { isGridDataEqual, isOrderedLayoutEqual } from '../../utils/equality_checks';
+import { resolveGridSection } from '../../utils/resolve_grid_section';
 import { getSensorType, isKeyboardEvent } from '../sensors';
+import { PointerPosition, UserInteractionEvent } from '../types';
+import { getDragPreviewRect, getResizePreviewRect, getSensorOffsets } from './utils';
+import { resolveSections } from '../../utils/section_management';
+
+let startingLayout: OrderedLayout | undefined;
 
 export const startAction = (
   e: UserInteractionEvent,
   gridLayoutStateManager: GridLayoutStateManager,
-  type: 'drag' | 'resize',
-  rowId: string,
+  sectionId: string,
   panelId: string
 ) => {
-  const panelRef = gridLayoutStateManager.panelRefs.current[rowId][panelId];
+  const panelRef = gridLayoutStateManager.panelRefs.current[panelId];
   if (!panelRef) return;
 
+  startingLayout = gridLayoutStateManager.gridLayout$.getValue();
   const panelRect = panelRef.getBoundingClientRect();
-
-  gridLayoutStateManager.interactionEvent$.next({
-    type,
+  gridLayoutStateManager.activePanelEvent$.next({
+    type: 'click',
```

Comment:
Because we combined `PanelInteractionEvent` and `ActivePanel` into `ActivePanelEvent`, the `--active` classname would get added the **instant** the mouse went down - which meant that "click" events were blocked by the `hasActivePanel` styles (specifically,  `pointerEvents: 'none'`). This meant that clicking the panel title could no longer open the settings flyout.

In order to allow click events through, we avoid adding this class until after the panel is actually **dragged**